### ISSUE TITLE
Youtube audio muting not working 


### ISSUE THREAD

**Issue by Imveryweird** [richardfrost/AdvancedProfanityFilter#258]
#### :twisted_rightwards_arrows: Steps To Reproduce
1. Go to youtube.com
2. Click on video with captions
3. Wait for the audio muting that never comes. The subtitles get shown censored but audio never gets muted

#### :computer: Details
 - **Device:** [samsung s8+]
 - **OS:** [``` android 8.0 ```]
 - **Browser:** [Firefox(fennec fdroid, icecat) [Latest Version Kiwi]
 - **Browser Version:** [ 68.12.0]

**Comment by richardfrost** [comment]
Can you send me a particular video, because I have tried 4 different videos on a desktop browser and they all seem to work as expected. There have been some new changes to YouTube recently, where they are censoring their auto-generated captions, which can make the filter miss muting when you would expect it to. See more information here: #251 

If those videos are working as expected on a desktop browser but not mobile, it will help me know to better know where to look to address the bug.

**Comment by Imveryweird** [comment]
https://www.youtube.com/watch?v=eQ_8F4nzyiw&app=desktop#dialog
Is one video that didn’t work.

**Comment by Imveryweird** [comment]
Man, i thought audio muting was on by default. *sigh* the problem was i didnt have the darn thing turned on, oh well.

**Comment by richardfrost** [comment]
Sorry about that! I took a poll when I added the audio muting feature, and it was decided it would be best to keep it as an opt-in feature since the original purpose of the filter was focused on text. Also, since the audio has a few caveats (captions/subtitles must be enabled, muting is only as good as the timing of the captions), I still think its best to leave it that way to hopefully provide time for users to read about the feature. I'm definitely open to hearing other opinions though.

**Comment by Imveryweird** [comment]
[This video](https://youtu.be/eQ_8F4nzyiw)from 12:05-12:15 has a wirty dord that is not censored on mobile.

**Comment by richardfrost** [comment]
Just to double check, did you have the captions enabled when you tested it? I just checked on Kiwi and it seemed to work fine. I'll try to do Firefox later tonight.

**Comment by Imveryweird** [comment]
Captions were on, maybe firefox is the issue.

**Comment by richardfrost** [comment]
Could you try switching the mute method (Options > Audio) to "Video volume"? Once you change that setting, please refresh the page and try again.

**Comment by Imveryweird** [comment]
Still isnt working, on kiwi either.

**Comment by richardfrost** [comment]
That's strange. Okay - are you viewing the page in mobile or desktop mode? I don't think it should make a difference, but it might be worth switching it. I was having problems with seeking in the video in mobile mode, so I switched to desktop to test it.

**Comment by Imveryweird** [comment]
That fixed it, but I had to switch to video volume.

**Comment by richardfrost** [comment]
Okay, I was being silly. The issue is that mobile YouTube uses a different domain (`m.youtube.com`), so the filter wasn't finding the required config to do muting. To fix that, add:

```json
{
  "m.youtube.com": {
    "mode": "element",
    "className": "caption-window",
    "subtitleSelector": "span.ytp-caption-segment",
    "tagName": "DIV"
  }
}
```

to your custom audio sites (Options > Audio) and click save. Then it should work with the mobile version. I will add this in next official release, at which point you could remove it (or don't add it, and just stick with the desktop mode).

As for the muting method, that is usually browser dependent. I believe the "tab" mode should work on Kiwi, but I'm not sure about Firefox. If you found one that works though, might as well stick with it.

**Comment by richardfrost** [comment]
This bug was fixed in version [2.11.0](https://github.com/richardfrost/AdvancedProfanityFilter/releases/tag/2.11.0). Feel free to respond and reopen this issue if you still have trouble related to this problem.
